### PR TITLE
events can be fetched based on place id, rather than fetching all eve…

### PR DESCRIPTION
**Pull request by lexidelorey**
…nts and checking each for correct location

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/SpecificLocationEventsServlet.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+      String yesCOVIDSafe = 
+          (String) entity.getProperty(Constants.COVID_SAFE_PARAM);
+      List<String> invitedAttendees = 
+          (List<String>) entity.getProperty(Constants.INVITED_ATTENDEES_PARAM);
+      List<String> goingAttendees = 
+          (List<String>) entity.getProperty(Constants.GOING_ATTENDEES_PARAM);
+      List<String> notGoingAttendees = 
+          (List<String>) entity.getProperty(Constants.NOT_GOING_ATTENDEES_PARAM);
+      String creator = 
+          (String) entity.getProperty(Constants.CREATOR_PARAM);
```
Comment (rglassett) [inline src/main/java/com/google/sps/servlets/SpecificLocationEventsServlet.java:64]: This looks duplicative of code in `EventServlet.java`, consider extracting a class dedicated to converting between stored Entity types and other types.

File: src/main/java/com/google/sps/servlets/SpecificLocationEventsServlet.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(events));
+  }
+
+  /** Determines if the event is current or past. */
+  private String eventCurrency(Date endDate, String timeZone) {
+    SimpleDateFormat formatter = new SimpleDateFormat("MM-dd-yyyy HH:mm z");
+    formatter.setTimeZone(TimeZone.getTimeZone(timeZone));
+    Date date = new Date();
+    Date currentDate = new Date();
```
Comment (rglassett) [inline src/main/java/com/google/sps/servlets/SpecificLocationEventsServlet.java:111]: Do you need two new instances of `Date`? Seems to me you could re-use one.

File: src/main/java/com/google/sps/servlets/SpecificLocationEventsServlet.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+import java.lang.*;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.TimeZone;
+import java.text.ParseException; 
+
+@WebServlet("/location-specific-events")
+public class SpecificLocationEventsServlet extends HttpServlet {
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```
Comment (rglassett) [inline src/main/java/com/google/sps/servlets/SpecificLocationEventsServlet.java:32]: I'd argue this should be a `doGet`. The action fetches a resource and doesn't do any mutations.

**Review by rglassett — APPROVED** [review]
Nice change!

File: src/main/java/com/google/sps/servlets/SpecificLocationEventsServlet.java (hunk 3 above)
Comment (lexidelorey) [reply src/main/java/com/google/sps/servlets/SpecificLocationEventsServlet.java:32]: I agree, initially I had written it as a doGet. However, I do not know how to pass parameters to a doGet function, and I need to pass the placeID param to the function. Is there a way to pass parameters to a doGet?

**Pull request merged**